add-on/TA-Demisto/bin/demisto_incident.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+#
+# This code was written by Demisto Inc.
+#
+
+import json
+import time
+import requests
+from requests import Request
+
+
+class DemistoIncident():
+    def __init__(self, logger):
+        """
+        :rtype: object
+        """
+        self.logger = logger
+
+    def create_incident(self, authkey, data, verify_req, search_query="", search_url="", ssl_cert_loc="",
+                        result=None, search_name=None, proxies=None, url=None):
+        """
+            This method is used to create incident in Demisto. It takes four arguments and all are mandatory:
+            @url: Demisto URL, its mandatory parameter.
+            @authkey: Requires parameter, used for authentication.
+            @data: Incident information,
+            @verify: Indicates if self signed certificates are allowed or not.
+        """
+
+        incident = self.create_incident_dictionary(data, search_query, search_url, result, search_name)
+
+        s = requests.session()
+
+        # url is either the selected Demisto server, or sending to all of the available servers
+        if not url:
+            url = data.get('demisto_server', '')
+
+        self.logger.debug("JSON data for the Incident=" + json.dumps(incident))
+
+        req = Request('POST', url + "/incident/splunkapp", data=json.dumps(incident))
+        prepped = s.prepare_request(req)
+
+        prepped.headers['Authorization'] = authkey
+        prepped.headers['Content-type'] = "application/json"
+        prepped.headers['Accept'] = "application/json"
+
+        if ssl_cert_loc:
```

Comment:
If user will disable the certificate check but will save path in config you will still try to use it, this is confusing.
If it's disabled should always be disabled